libparsec/crates/types/src/local_device.rs

```diff
@@ -60,13 +62,29 @@ impl LocalDevice {
             signing_key: signing_key.unwrap_or_else(SigningKey::generate),
             private_key: private_key.unwrap_or_else(PrivateKey::generate),
             initial_profile,
-            user_realm_id: VlobID::default(),
-            user_realm_key: SecretKey::generate(),
+            user_realm_id: user_realm_id.unwrap_or_default(),
+            user_realm_key: user_real_key.unwrap_or_else(SecretKey::generate),
             local_symkey: SecretKey::generate(),
             time_provider: time_provider.unwrap_or_default(),
         }
     }
 
+    /// generates a new device for a given user
+    pub fn from_existing_device_for_user(
+        device: &LocalDevice,
+        device_label: DeviceLabel,
+    ) -> LocalDevice {
+        LocalDevice {
+            // device specif fields
+            device_label,
+            device_id: DeviceID::default(),
+            signing_key: SigningKey::generate(),
+            time_provider: TimeProvider::default(),
+            // the other fields are user related
+            ..device.clone()
+        }
```

Comment:
I would have use the method `LocalDevice::generate_new_device` instead, so we only have a single place where the field of `LocalDevice` are initialized

Reply:
I'd rather keep it that way, as I find it highlights better which fields are generated and which are the same.